### PR TITLE
[FLINK-30161] Add table schema validation for full compaction

### PR DESCRIPTION
**Pull request by zjureel**
The full compaction producer is not supported in `ValueCountMergeFunction`, it should be checked in `SchemaManager`

### CODE REVIEW

**Review by JingsongLi — COMMENTED** [review]
Thanks @zjureel !

I find that all kinds of verifications are a bit messy. Can we propose a verification method (Maybe a static method in `CoreOptions`)? It can be called in two places:
1. `SchemaManager.commit`: When submitting new schema changes.
2. `FileStoreTableFactory.create`: When creating a new table, the verification here is because there may be dynamic options, which also need to be verified.

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/CoreOptions.java [hunk 1]
```diff
@@ -695,6 +691,44 @@ public InlineElement getDescription() {
...
 
+    /**
+     * Validate the {@link TableSchema} and {@link CoreOptions}.
+     *
+     * <p>TODO validate all items in schema and all keys in options.
+     *
+     * @param schema the schema to be validated
+     * @param options the options to be validated
+     */
+    public static void validateTableOptions(TableSchema schema, CoreOptions options) {
```
Comment (JingsongLi) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/CoreOptions.java:702]: Minor: this options can generated by `schema`. Just `validateTableOptions(TableSchema schema)` is OK.

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/table/FileStoreTableFactory.java [hunk 2]
```diff
@@ -63,7 +62,7 @@ public static FileStoreTable create(
         newOptions.set(PATH, tablePath.toString());
 
         // validate merged options
-        validateOptions(new CoreOptions(newOptions));
+        CoreOptions.validateTableOptions(tableSchema, new CoreOptions(newOptions));
```
Comment (JingsongLi) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/table/FileStoreTableFactory.java:65]: We could validate new tableSchema. (It's down there)

**Review by JingsongLi — COMMENTED** [review]
Thanks! Left minor comments.

**Comment by zjureel** [comment]
Thanks @JingsongLi  Done

**Review by JingsongLi — APPROVED** [review]
Looks good to me!

**Pull request merged**